Src/VimWpf/Implementation/RelativeLineNumbers/Util/SafeRefreshLock.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Vim.UI.Wpf.Implementation.RelativeLineNumbers.Util
+{
+    public class SafeRefreshLock

```

Comment:
Know this code is from the original PR but I can't really discern the intent of this type. It seems over complicated. Seems like we could just use a simple `SemaphoreSlim` here or similar.

Reply:
Well, having fixed a few minor issues, I know know how to write the pure synchronous version and so we could make the async version still supported but not enabled.  I did not drill into the logic of the "safe lock" but I agree that I would rather risk a small performance hit that lingering fears that a bug lurks in there somewhere.

That being said, the code that runs in this lock is side-effect free and operates on immutable snapshots, and so I think that the worst that could happen is that it calculates two units of work in parallel. To put it another way, the lock is for preventing wasted CPU cycles, not for ensuring correctness.

Reply:
So daily use testing showed that the whole async strategy, if effective, needs to be done more carefully. I converted the margin to be entirely synchronous.

Reply:
The code necessary to make it async is small, so I deleted it. It'll be in the git history if we ever need to revive it.